packages/next/build/index.ts

```diff
@@ -361,7 +363,14 @@ export default async function build(dir: string, conf = null): Promise<void> {
           })
           staticCheckSema.release()
 
-          if (result.static && customAppGetInitialProps === false) {
+          if (result.prerender === true) {
+            sprPages.add(page)
+          }
+
+          if (
+            (result.static && customAppGetInitialProps === false) ||
+            result.prerender === 'legacy'
```

Comment:
Let's drop legacy for now -- seems it'd be trivial to re-add but I want to have an RFC first.